tianshou/data/utils/converter.py

```diff
@@ -6,67 +6,71 @@
 from tianshou.data.batch import _parse_value, Batch
 
 
-def to_numpy(x: Union[
-    Batch, dict, list, tuple, np.ndarray, torch.Tensor]) -> Union[
-        Batch, dict, list, tuple, np.ndarray, torch.Tensor]:
+def to_numpy(
+    x: Optional[Union[Batch, dict, list, tuple, np.number, np.bool_, Number,
+                      np.ndarray, torch.Tensor]]
+) -> Union[Batch, dict, list, tuple, np.ndarray]:
     """Return an object without torch.Tensor."""
     if isinstance(x, torch.Tensor):  # most often case
-        x = x.detach().cpu().numpy()
+        return x.detach().cpu().numpy()
     elif isinstance(x, np.ndarray):  # second often case
-        pass
+        return x
     elif isinstance(x, (np.number, np.bool_, Number)):
-        x = np.asanyarray(x)
+        return np.asanyarray(x)
     elif x is None:
-        x = np.array(None, dtype=np.object)
+        return np.array(None, dtype=np.object)
     elif isinstance(x, Batch):
         x.to_numpy()
+        return x
```

Comment:
I think this inplace operation is a bug.

Reply:
I don't know, it is hard to tell. But in some case, no copy is required and in-place is more efficient. But at least it should be documented.